### PR TITLE
ADEN-14010 :: Add FandomTaxonomy consumer

### PR DESCRIPTION
**Pull request by vforge**
- Add consumer for https://github.com/Wikia/unified-platform/pull/17211
- Replace current DTCO hack logic with proper one

### CODE REVIEW

File: src/core/services/fandom-taxonomy-service.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	public static applyToTargetingService() {
+		const fandomTaxonomy = new FandomTaxonomy();
+
+		// Regular Key Values are just added to the targeting service
+		targetingService.extend({
+			...targetingService.dump(),
+			...fandomTaxonomy.getKeyValues(),
+		});
+
+		// Tags are added to the targeting service as well, but should be merged with the existing values (if possible)
```
Comment (szymon-nawrocki) [inline src/core/services/fandom-taxonomy-service.ts:26]: Could you tell me why the taxonomy tags are added to new `fandomTaxonomy` object?
We already have taxonomy tags added to `window.fandomContext.page.tags` (page level tags) and `window.fandomContext.site.tags` (site level tags) and these are already 'existing' values that you are mentioning in this comment.

I am not the biggest fan of the fact that we have taxonomy tags in 2 separate objects and two of these objects are used by ad-engine for getting the same thing - taxonomy tags

Comment (vforge) [reply src/core/services/fandom-taxonomy-service.ts:26]: @szymon-nawrocki This new taxonomy will be injected into the page via the JavaScrtipt script tag. The intention is not to pollute any existing object and not have conflict with it.

As for the long-term plans for current tags, IIRC the ad-tag-manager-based tags will the replaced by DTCO tags, but I am not sure what will happen to the rest of them. This is the current requirement from Eric to have it work that way.

Comment (vforge) [reply src/core/services/fandom-taxonomy-service.ts:26]: For some of the new DTCO values we're creating new KVs (code above), but for `fran` tag specifically, we were asked to merge it with the existing one. The current code is my best attempt at doing that in a clear way, but I am open to any other suggestions.

Comment (vforge) [reply src/core/services/fandom-taxonomy-service.ts:26]: I am sure the long-term intention is to replace the current page/wiki tags with DTCO taxonomy, but for now, we need to have some overlap.

Comment (szymon-nawrocki) [reply src/core/services/fandom-taxonomy-service.ts:26]: okay, thanks for the explanation

**Review by szymon-nawrocki — APPROVED** [review]

**Pull request merged**